torch/multiprocessing/cuda_multiprocessing.md

```diff
@@ -0,0 +1,79 @@
+# CUDA IPC Refcounting implementation explained
+
+Since shared CUDA memory belongs to the producer process, we need to take special precautions to make sure that it is stays allocated for entire shared tensor life-span.
+
+It could be done manually by syncing on an event:
+
+```python
+# Producer
+queue.put(tensor)
+event.wait()
+
+# Consumer
+tensor = queue.get()
+safe_to_use_tensor = tensor.clone()
+event.set()
+```
+
+However, this requires additional memory as well as blocking producer process (and gets overcomplicated in case of multiple consumers and handling various race-conditions).
```

Comment:
It doesn't really require additional memory, does it? The bigger problem was people would forget to do this and WHAMMO memory corruption.

Reply:
Well additional memory option is:
  x = torch.zeros(10)
  queue.put(x)
  keep_all_tensors.append(x)

Reply:
Oh sure, if you just leak the memory then that also solves the problem, but it didn't sound like that was what you were talking about here ;)